packages/matter.js/src/CommissioningController.ts

```diff
@@ -122,7 +125,7 @@ export class CommissioningController extends MatterNode {
     /** Internal method to initialize a MatterController instance. */
     private async initializeController() {
         if (this.mdnsScanner === undefined || this.storage === undefined) {
-            throw new ImplementationError("Add the node to the Matter instance before!");
+            throw new ImplementationError("Add the node to the Matter instance before.");
```

Comment:
OMG this diff would be a lot shorter if we were validating state in helper methods.  Could we do this now?

E.g. something like assertParented(this), assertConnected(this), assertStarted(this), etc.

You can assert "mdnsScanner" and "storage" are both defined with a single method too, [I made an example](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQHZgLYFMAuaCAFwCcBLLAc2gF5oAiWc-AEyYG51MB3fJRoALUgH5iWAK64ARvnI9M0MDXwTsM+YtQBfVKgBmUrMFKV4WFVAWkAClQBuYYAE8AahfCkOACkOIxAjwAJTEkBC2MAFIlDDB0ABkKNACQqKSWgoANCpqmXIK0LoovBiUhtD+iAB0aSKkDPSMJuz4htQc0AA+3dAxNar4TS1YbR1YHCGlysqkwuTwfNj4ywCi5IvkviFKmPr6qMCWZP2IDCvLwTs8x1insmDsAOqCDQXaFwP1orcnjY92ABBfKaQrkL61IY8VARWwOSjONyeeDePwxXZHf7QGiIF5vDJgz6Mb4E0h-e6NXHwYGg6TgyHwQZqHhAA)

Reply:
This works if the properties are public, but not when they are private (as they are currently)

Reply:
So if you have a idea working without making that public I'm happy to adjust ... I found nothing ... I also do not like it but here we lack the Typescript support for that :-(